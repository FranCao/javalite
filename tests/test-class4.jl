class Record {
    int index;
    int data;
    string comment;
}

void updateComment(class Record r, string cmt) {
    string msg = concat("old comment: ", r.comment);
    print(msg);
    r.comment = cmt;
}

int main() {
    class Record rec = Record(0, 42, "added on Apr 10, 2021");
    updateComment(rec, "updated on Apr 18, 2021");
    string msg = concat("new comment: ", rec.comment);
    print(msg);
}