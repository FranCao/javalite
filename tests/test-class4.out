old comment: added on Apr 10, 2021
new comment: updated on Apr 18, 2021